pkg/metric_generator/allow_labels.go

```diff
@@ -0,0 +1,40 @@
+package generator
+
+import "regexp"
+
+var denyAllByDefaultMetrics = regexp.MustCompile(".*_labels$|.*_annotations$")
+
+type AllowLabels interface {
+	Allowed(metric string, labels, values []string) (allowedLabels, allowedValues []string)
+}
+
+type allowLabels struct {
+	allow map[string][]string
+}
+
+func NewAllowLabels(allow map[string][]string) AllowLabels {
+	return &allowLabels{
+		allow: allow,
+	}
+}
+
+func (a *allowLabels) Allowed(metric string, labels, values []string) ([]string, []string) {
+	var allowedLabels, allowedValues []string
+	allowed, ok := a.allow[metric]
+	if !ok{
+		if denyAllByDefaultMetrics.MatchString(metric){
```

Comment:
I'm a bit worried about the consequence this might have on performance if we do this on every metric. Let's make sure that this doesn't impact the benchmarks very much.